kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt

```diff
@@ -115,6 +120,35 @@ fun <InputT, OutputT : Any, RenderingT, RunnerT> launchWorkflowIn(
     beforeStart = beforeStart
 )
 
+/**
+ * Launches the [workflow] in a new coroutine in [scope]. The workflow tree is seeded with
+ * [initialSnapshot] and the first value emitted by [inputs].  Subsequent values emitted from
+ * [inputs] will be used to re-render the workflow.
+ *
+ * Like [launchWorkflowIn], but will automatically cancel the workflow as soon as the first
+ * [output][OutputT] is emitted. [beforeStart] gets the same [Flow] of [RenderingAndSnapshot], but
+ * instead of a flow of outputs it gets a [Deferred] of the first output.
+ */
+@UseExperimental(ExperimentalCoroutinesApi::class)
+fun <InputT, OutputT : Any, RenderingT, ResultT> launchWorkflowUntilFirstOutputIn(
```

Comment:
I kind of hate this name. `launchSingleOutputWorkflowIn`? `launchResultWorkflowIn`?

Reply:
I like `launchSingleOutputWorkflowIn`.

![giphy](https://user-images.githubusercontent.com/1884445/61164781-2d7e1780-a4cd-11e9-824c-0383367fb109.gif)